pkg/reconciler/route/resources/virtual_service.go

```diff
@@ -199,81 +191,128 @@ func buildGatewayName(gatewayName string) (string, error) {
 	return gatewayName, nil
 }
 
+type httpRoutesBuilder struct {
+	routes               v1alpha1.RouteSpecFieldsSlice
+	appBindings          map[string]RouteBindingSlice
+	routeServiceBindings map[string][]v1alpha1.RouteServiceDestination
+	defaultConfig        *kfconfig.DefaultsConfig
+}
+
+func newHTTPRoutesBuilder(
+	routes v1alpha1.RouteSpecFieldsSlice,
+	appBindings map[string]RouteBindingSlice,
+	routeServiceBindings map[string][]v1alpha1.RouteServiceDestination,
+	defaultConfig *kfconfig.DefaultsConfig,
+) *httpRoutesBuilder {
+	return &httpRoutesBuilder{
+		routes:               routes,
+		appBindings:          appBindings,
+		routeServiceBindings: routeServiceBindings,
+		defaultConfig:        defaultConfig,
+	}
+}
+
+func (hb *httpRoutesBuilder) appBindingsFor(rsf v1alpha1.RouteSpecFields) RouteBindingSlice {
```

Comment:
nit: should this be named 'routeBindingSliceFor'?

Reply:
Done, thanks